### ISSUE TITLE
Could not compile dependency ssl_verify_hostname

### ISSUE THREAD

**Issue by jguyon** [edgurgel/httpoison#46]
I added httpoison to my mix.exs, but when I run `mix deps.compile`, I get the following error:

```
==> idna (compile)
==> ssl_verify_hostname (compile)
Compiling src/ssl_verify_hostname.erl failed:
src/ssl_verify_hostname.erl:8: can't find include lib "public_key/include/public_key.hrl"
src/ssl_verify_hostname.erl:30: undefined macro 'id-ce-subjectAltName'
src/ssl_verify_hostname.erl:20: record 'AttributeTypeAndValue' undefined
src/ssl_verify_hostname.erl:22: variable 'CN' is unbound
src/ssl_verify_hostname.erl:170: record 'OTPTBSCertificate' undefined
src/ssl_verify_hostname.erl:175: record 'OTPCertificate' undefined
src/ssl_verify_hostname.erl:177: function extract_dns_names/1 undefined
src/ssl_verify_hostname.erl:38: Warning: function extract_dns_names_from_alt_names/2 is unused
ERROR: compile failed while processing /home/jerome/Development/soundcloud/deps/ssl_verify_hostname: rebar_abort
** (Mix) Could not compile dependency ssl_verify_hostname, /home/jerome/.mix/rebar command failed. If you want to recompile this dependency, please run: mix deps.compile ssl_verify_hostname
```

Do I need to install a dependency or something ?

**Comment by edgurgel** [comment]
Hi @jguyon, this seems to be a problem with missing headers from erlang. If you are using ubuntu you may need to install `erlang-src` or something like this. The error mentions `public_key.hrl` which is part of erlang header files.

**Comment by jguyon** [comment]
Thanks for your response @edgurgel.
I forgot to mention it, but I do have `erlang-src` installed. Is there any other package I could be missing?

**Comment by jguyon** [comment]
So I searched for `erlang` in apt to have an idea of which package could be missing, and I solved my problem by installing `erlang-dev`. Thanks again for your response.

**Comment by mgwidmann** [comment]
:+1:

**Comment by nettofarah** [comment]
Would it make sense to add a note in the Readme file for ubuntu users? Or is this something that we'd see on any elixir library? assuming you're running ubuntu and haven't installed `erlang-dev` yet.

**Comment by araslanov-e** [comment]
:+1:

**Comment by nin042** [comment]
:+1:

**Comment by venadHD** [comment]
This solution is viable for this error too:

`src/ssl_verify_pk.erl:16: can't find include lib "public_key/include/public_key`

Added for googleners

**Comment by trans** [comment]
Hmm... I have both of these installed (`erlang-dev` and `erlang-src`) and still get the error. 

Could it be because I installed Erlang via deb package but installed Elixir via source in `opt/`? Is so how to fix?

**Comment by cognivore** [comment]
I've noticed this on `1.15.0-dev` too!

**Comment by axelson** [comment]
@cognivore that's likely a slightly different error (:ssl_verify_fun vs :ssl_verify_hostname). If it is the `:ssl_verify_fun` issue then you can fix it with this in your mix.exs `deps` function:
```
# Temporarily set the manager option for this so it compiles
# https://elixirforum.com/t/elixir-v1-15-0-released/56584/4?u=axelson
{:ssl_verify_fun, ">= 0.0.0", manager: :rebar3, override: true}
```

**Comment by harryloewen** [comment]
Fixed in ssl_verify_fun [1.1.7](https://github.com/deadtrickster/ssl_verify_fun.erl/releases/tag/1.1.7) :tada:
Relevant pr https://github.com/deadtrickster/ssl_verify_fun.erl/pull/27
